packages/happy-dom/src/nodes/html-dialog-element/HTMLDialogElement.ts

```diff
@@ -65,15 +66,31 @@ export default class HTMLDialogElement extends HTMLElement {
 		this.removeAttribute('open');
 		this.returnValue = returnValue;
 		if (wasOpen) {
+			this.#removeKeydownListener();
 			this.dispatchEvent(new Event('close'));
 		}
 	}
 
+	#keydownHandler = (event: KeyboardEvent): void => {
+		if (event.key === 'Escape') {
```

Comment:
In order to support clicking a button, we might be able to override `dispatchEvent()` and check if the event is "click" and the target is a button or input with type button.